url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/visualisation/Generic_Visualisation.java

```diff
@@ -0,0 +1,289 @@
+                //} catch (IOException e) {
+                //} catch (Exception e) {
+                // This can happen because of too many open files.
+                // Try waiting for a 2 seconds and then repeating...
+                try {
+                    synchronized (aFile) {
+                        aFile.wait(2000L);
+                    }
+                } catch (InterruptedException ex) {
+                    Logger.getLogger(Generic_Visualisation.class.getName()).log(Level.SEVERE, null, ex);
+                }
+                return loadImage(aFile);
+            }
+        } catch (IOException e) {
+            Generic_Log.logger.log(
+                    Generic_Log.Generic_DefaultLogLevel, //Level.ALL,
+                    e.getMessage());
+            e.printStackTrace(System.err);
+            System.exit(Generic_ErrorAndExceptionHandler.IOException);
+        }
+        return aBufferedImage;
+    }
+
+    /**
+     * For saving a BufferedImage to a File
+     *
+     * @param bi
+     * @param format
+     * @param aFile
+     */
+    public static void saveImage(
+            BufferedImage bi,
+            String format,
+            File aFile) {
+//        try {
+        System.out.println("<Generic_Visualisation.saveImage(BufferedImage,String,File)>");
+        if (bi != null) {
+            try {
+                System.out.println("File " + aFile.toString());
+                ImageIO.write(bi, format, aFile);
+            } catch (IIOException e) {
+                System.err.println("Trying to handle " + e.getLocalizedMessage());
+                System.err.println("Wait for 2 seconds then trying again to saveImage.");
+                //e.printStackTrace(System.err);
+                // This can happen because of too many open files.
+                // Try waiting for 2 seconds and then repeating...
+                try {
+                    synchronized (bi) {
+                        bi.wait(2000L);
+                    }
+                } catch (InterruptedException ex) {
+                    Logger.getLogger(Generic_Visualisation.class.getName()).log(Level.SEVERE, null, ex);
+                }
+                saveImage(
+                        bi,
+                        format,
+                        aFile);
+                //} catch (FileNotFoundException e) {
+                //} catch (IOException e) {
+            } catch (Exception e) {
+                System.err.println("Trying to handle " + e.getLocalizedMessage());
+                System.err.println("Wait for 2 seconds then trying again to saveImage.");
+                //e.printStackTrace(System.err);
+                // This can happen because of too many open files.
+                // Try waiting for 2 seconds and then repeating...
+                try {
+                    synchronized (bi) {
+                        bi.wait(2000L);
+                    }
+                } catch (InterruptedException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)